crates/sources/src/scenarios/mod.rs

```diff
@@ -28,12 +28,15 @@ mod test {
             tables.fetches = tables::Fetches::new();
             tables.resources = tables::Resources::new();
 
-            // Clear implicit imports, leaving only explicit catalog imports.
-            tables.imports.retain(|import| import.scope.fragment().unwrap().starts_with("/import"));
-
             // Verify shape of inline specs.
 			insta::assert_debug_snapshot!(tables);
 
+            // Most implicit imports were removed. Some remain:
+            // * References between JSON schemas (but not from a collection to a schema).
+            // * Imports which were not found and thus could not be inlined.
+            // Clear remaining implicit imports, leaving only explicit catalog imports.
+            tables.imports.retain(|import| import.scope.fragment().unwrap().starts_with("/import"));
```

Comment:
Not changed in this PR, but just curious: I wonder if it would make sense to explicitly remove all imports that match the conditions mentioned (inter-JSON-schema references, imports that were not found), and then assert that there are no more unexplained imports left? It's probably not a big deal since this is working (and is tested), but handling the expected and then asserting that nothing unexpected remains seems like a good practice.

Reply:
:shrug: that would be fine. in development, i did want to see them in snapshots, which is why i recordered this `retain` -- to be able to confirm the presence/absence of specific expected items.

TBH I don't recall exactly why I didn't just clear _all_ imports here. I left the retain of explicit imports simply to avoid churning snapshots more than I had, and because I didn't have a good reason to take it away.